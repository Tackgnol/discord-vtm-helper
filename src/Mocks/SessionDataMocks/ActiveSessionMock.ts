import { IActiveSession } from '../../Models/AppModels';
import { existingGTQueryWithoutValue } from '../QueryMocks/globalTestQueries';

export const activeSessionMock: IActiveSession = {
	channelId: '0',
	gameId: '0',
	prevCommand: existingGTQueryWithoutValue,
};
