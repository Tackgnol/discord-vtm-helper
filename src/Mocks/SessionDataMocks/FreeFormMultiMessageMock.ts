export const freeFormMultiMessageMock = {
	reply: 'test',
	userList: ['1', '2', '3'],
	channelMembers: ['2', '3', '4'],
};

export const freeFormMultiMessageErrorMock = {
	reply: 'test',
	userList: ['1', '2', '3'],
	channelMembers: ['4', '5', '6'],
};
