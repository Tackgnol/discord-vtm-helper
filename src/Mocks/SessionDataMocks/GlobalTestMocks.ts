import { GlobalTest } from '../../Models/GameData';

export const globalTestWithoutShortCircuitMock: GlobalTest = {
	testMessage: 'test',
	shortCircuit: false,
	globaltestoptionSet: [
		{ minResult: 1, resultMessage: 'result for 1' },
		{ minResult: 5, resultMessage: 'result for 5' },
	],
	name: 'test',
	replyPrefix: 'testPrefix',
};
export const globalTestWithShortCircuitMock: GlobalTest = {
	testMessage: 'shortCircuitTest',
	shortCircuit: true,
	globaltestoptionSet: [
		{ minResult: 1, resultMessage: 'result for 1' },
		{ minResult: 5, resultMessage: 'result for 5' },
	],
	name: 'scTest',
	replyPrefix: 'testPrefix',
};
