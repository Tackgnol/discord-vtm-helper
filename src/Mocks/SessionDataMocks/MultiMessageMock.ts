import { IMultiPlayerMessage } from '../../Models/GameData';

export const multiMessageMock: IMultiPlayerMessage = {
	userList: ['0'],
	name: 'test',
	message: "you've got mail",
};
