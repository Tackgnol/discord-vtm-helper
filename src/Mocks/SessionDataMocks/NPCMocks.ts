import { NPC } from '../../Models/GameData';

export const testNPCWithFacts: NPC = {
	callName: 'testWithFacts',
	description: 'this is a test',
	image: 'placeholder',
	name: 'Test',
	facts: ['fact1', 'fact2'],
	gameId: 'x',
};

export const testNPCWithoutFacts: NPC = {
	callName: 'testWithout',
	description: 'this is a test',
	image: 'placeholder',
	name: 'Test',
	facts: [],
	gameId: 'x',
};
