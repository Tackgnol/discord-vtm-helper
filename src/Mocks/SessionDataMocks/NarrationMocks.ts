import { INarration } from '../../Models/GameData';

export const basicNarration: INarration = {
	name: 'test',
	image: 'urlToImage',
	narrationText: 'this is a narration text',
};
