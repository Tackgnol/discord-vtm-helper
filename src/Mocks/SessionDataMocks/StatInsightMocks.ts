import { IStatInsight } from '../../Models/GameData';

export const statInsightMock: IStatInsight = {
	statName: 'testStat',
	successMessage: 'winner is you!',
	minValue: 2,
	name: 'test',
};
