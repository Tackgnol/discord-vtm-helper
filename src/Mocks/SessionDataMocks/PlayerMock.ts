import { Player } from '../../Models/GameData';
import { testNPCWithFacts, testNPCWithoutFacts } from './NPCMocks';

export const playerMock: Player = {
	npcSet: [testNPCWithFacts, testNPCWithoutFacts],
	discordUserName: 'test',
	statisticsSet: [{ name: 'test', value: 2 }],
	id: '0',
};
