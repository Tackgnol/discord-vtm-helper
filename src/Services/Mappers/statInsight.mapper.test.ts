import chai, { expect } from 'chai';
import { jestSnapshotPlugin } from 'mocha-chai-jest-snapshot';
import { statInsightMockHigh } from '../../Mocks/SessionDataMocks/StatInsightMocks';
import { statInsightMapper } from './statInsight.mapper';

chai.use(jestSnapshotPlugin());

describe('Services >> Mappers >> statInsightMapper', () => {
	it('properly maps a valid statInsight', () => {
		const result = statInsightMapper(statInsightMockHigh);
		expect(result).toMatchSnapshot();
		expect(result).to.exist;
	});

	it('throws on name missing', () => {
		expect(() => {
			statInsightMapper({ ...statInsightMockHigh, name: undefined });
		}).to.throw(Error);
	});
	it('throws on statName missing', () => {
		expect(() => {
			statInsightMapper({ ...statInsightMockHigh, statName: undefined });
		}).to.throw(Error);
	});

	it('throws on successMessage missing', () => {
		expect(() => {
			statInsightMapper({ ...statInsightMockHigh, successMessage: undefined });
		}).to.throw(Error);
	});

	it('throws on minValue missing', () => {
		expect(() => {
			statInsightMapper({ ...statInsightMockHigh, minValue: undefined });
		}).to.throw(Error);
	});
});
