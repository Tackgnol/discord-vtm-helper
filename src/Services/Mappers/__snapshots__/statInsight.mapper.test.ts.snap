// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services >> Mappers >> statInsightMapper properly maps a valid statInsight 1`] = `
Object {
  "minValue": 5,
  "name": "test",
  "statName": "test",
  "successMessage": "winner is you!",
}
`;
